### PR TITLE
Integration tests for deviceGroups in PcsConfig and wait for SeedComplete

### PR DESCRIPTION
**Pull request by isaac73**
# Description and Motivation 
Integration tests for deviceGroups in PcsConfig
...

# Change type 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [x] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: Helper/SeedData.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+{
+    public static class SeedData
+    {
+        private const int TIMEOUT = 10000;
+        private const int RETRYLIMIT = 10;
+        private static HttpClient httpClient = new HttpClient();
+
+        public static void WaitForSeedComplete()
+        {
+            bool isSeedComplete = false;
```
Comment (molly-moen) [inline Helper/SeedData.cs]: is the expectation here that if seed data doesn't complete within the allotted time/retries the tests will continue on? Shouldn't they fail?

Comment (isaac73) [reply Helper/SeedData.cs]: The tests will continue. But they will eventually fail, since there is no seeded data.

Comment (molly-moen) [reply Helper/SeedData.cs]: why not preemptively fail with a nice message so it's clear what went wrong?

File: Helper/SeedData.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿using Helpers.Http;
```
Comment (ppathan) [inline Helper/SeedData.cs]: Is SeedData.cs being used by any other projects? If not move it to PcsConfig

File: PcsConfig/DeviceGroupsTests.cs [hunk 3]
```diff
@@ -0,0 +1,280 @@
+﻿using Helpers.Http;
```
Comment (ppathan) [inline PcsConfig/DeviceGroupsTests.cs]: nit: Remove Pcs from the file names and folder names

File: Helper/SeedData.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+using System.Net;
+
+namespace Helpers
+{
+    public static class SeedData
+    {
+        private const int TIMEOUT = 10000;
```
Comment (ppathan) [inline Helper/SeedData.cs]: nit: TIMEOUT_MS

File: PcsConfig/DeviceGroupsTests.cs [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+            request.Options.Timeout = TIMEOUT;
+
+            // Act for create
+            var response = this.httpClient.PostAsync(request).Result;
+
+            // Assert for create
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var createdDeviceGroup = JsonConvert.DeserializeObject<DeviceGroupApiModel>(response.Content);
+            Assert.Equal(inputDeviceGroup.DisplayName, createdDeviceGroup.DisplayName);
+            Assert.Equal(inputDeviceGroup.Conditions.ToString(), createdDeviceGroup.Conditions.ToString());
```
Comment (ppathan) [inline PcsConfig/DeviceGroupsTests.cs]: Move the the Create to a private method and call it from `ShouldCreateNewDeviceGroup` and `ShouldUpdateCreatedDeviceGroup`

File: PcsConfig/DeviceGroupsTests.cs [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+
+            // Dispose
+            disposeList.Add(createdDeviceGroup.Id);
+
+            // Arange for create 2
+            inputDeviceGroup.DisplayName = "TEST - CreateDeviceGroup - 2";
```
Comment (ppathan) [inline PcsConfig/DeviceGroupsTests.cs]: When you create a private method that creates device group pass in DisplayName as a parameter so you can do multiple creates as well

File: Helper/SeedData.cs (hunk 2 above)
Comment (isaac73) [reply Helper/SeedData.cs]: @jillcary Do you need it? I suppose yes.

File: PcsConfig/DeviceGroupsTests.cs (hunk 3 above)
Comment (isaac73) [reply PcsConfig/DeviceGroupsTests.cs]: done

File: Helper/SeedData.cs (hunk 4 above)
Comment (isaac73) [reply Helper/SeedData.cs]: done

File: PcsConfig/DeviceGroupsTests.cs (hunk 5 above)
Comment (isaac73) [reply PcsConfig/DeviceGroupsTests.cs]: done

File: PcsConfig/DeviceGroupsTests.cs (hunk 6 above)
Comment (isaac73) [reply PcsConfig/DeviceGroupsTests.cs]: done

File: Config/SeedDataTest.cs [hunk 7]
```diff
@@ -4,21 +4,23 @@
...
-        private const string CONFIG_ADDRESS = "http://127.0.0.1:9005/v1";
-        private const string TELEMETRY_ADDRESS = "http://127.0.0.1:9004/v1";
-        private const string SIMULATION_ADDRESS = "http://127.0.0.1:9003/v1";
+        private const int TIMEOUT_MS = 10000;
+        private bool seedComplete = false;
 
         public SeedDataTest()
         {
-            this.httpClient = new HttpClient();
+            httpClient = new HttpClient();
```
Comment (molly-moen) [inline Config/SeedDataTest.cs]: this.httpClient

File: Config/SeedDataTest.cs [hunk 8]
```diff
@@ -42,16 +44,17 @@ public void Should_Return_OK_Status()
...
         /// is created.
         /// </summary>
-        [Fact, Trait("Type", "IntegrationTest")]
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
         public void DeviceGroupsAreCreatedOnStartup()
         {
-            // Arrange
-            // wait for config to run seed data
-            System.Threading.Thread.Sleep(60000);
+            if (!seedComplete)
```
Comment (molly-moen) [inline Config/SeedDataTest.cs]: you can just do Assert.True(seedComplete, "Seed data failed")

File: scripts/containers.sh [hunk 9]
```diff
@@ -7,19 +7,19 @@ APP_HOME="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && cd .. && pwd )/"
 cd $APP_HOME
 
 start() {
-    ./scripts/pcsconfig.sh start
     ./scripts/storageadapter.sh start
     ./scripts/devicesimulation.sh start
     ./scripts/telemetry.sh start
+    ./scripts/config.sh start
```
Comment (molly-moen) [inline scripts/containers.sh]: is there any issue with moving config down?

Comment (isaac73) [reply scripts/containers.sh]: Config needs to be started after the containers above it, in order to successfully complete seeding on startup. Though we have 10 retry with 10 sec interval for seeding, it's better this way.

File: Config/DeviceGroupsTests.cs [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+    [Collection("Config Tests")]
+    public class DeviceGroupsTests : IDisposable
+    {
+        private readonly IHttpClient httpClient;
+        private const int TIMEOUT_MS = 10000;
+        private List<string> disposeList = new List<string>();
+
+        public DeviceGroupsTests()
+        {
+            httpClient = new HttpClient();
```
Comment (molly-moen) [inline Config/DeviceGroupsTests.cs]: nit: this.

File: Config/DeviceGroupsTests.cs [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+            };
+            var request = new HttpRequest();
+            request.SetContent(inputDeviceGroup);
+            request.SetUriFromString(Constants.CONFIG_ADDRESS + "/devicegroups");
+            request.Options.Timeout = TIMEOUT_MS;
+
+            // Act
+            var response = this.httpClient.PostAsync(request).Result;
+
+            // AssertÏ
```
Comment (molly-moen) [inline Config/DeviceGroupsTests.cs]: nit: weird character after assert

File: Config/DeviceGroupsTests.cs (hunk 10 above)
Comment (isaac73) [reply Config/DeviceGroupsTests.cs]: done

File: Config/DeviceGroupsTests.cs (hunk 11 above)
Comment (isaac73) [reply Config/DeviceGroupsTests.cs]: done

**Pull request merged**